### PR TITLE
Integrate Touchstone login

### LINKED ISSUE #62: Add auth library and implement Touchstone login
**Depends on common auth library: https://github.com/mitodl/ol-django/issues/3**

We have a Touchstone login implementation that should be able to be ported over to fit into the auth library pipeline with some config changes.

**Acceptance criteria:**
- ol-django auth library added as a dependency
- Touchstone login integrated with authentication pipeline
- Simple Touchstone login page added

### PR DESCRIPTION
**Pull request by rhysyngsun**
#### Pre-Flight checklist

- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested
- [x] Settings
  - [x] New settings are documented and present in `app.json`
  - [x] New settings have reasonable development defaults, if applicable
  - [x] Opened issue for DevOps regarding necessary configuration changes to deployed environments

#### What are the relevant tickets?
Closes #62 

#### What's this PR do?
This adds integration with Touchstone to enable user account creation/login.

#### How should this be manually tested?
Follow the instructions in the README for setting up your local environment w/ SSOCircle and verify you can login w/ it.

### CODE REVIEW

**Review by mbertrand — CHANGES_REQUESTED** [review]
Getting this error after returning from SSOCircle to ocw-studio:
```
web_1     | django.urls.exceptions.NoReverseMatch: Reverse for 'login' not found. 'login' is not a valid view function or pattern name.
```

Full stacktrace:
```
nginx_1   | 172.31.0.1 - - [01/Apr/2021:13:51:24 +0000] "GET /login/saml/?idp=default HTTP/1.1" 302 0 "http://studiomb.odl.local:8043/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36" "-"
web_1     | [2021-04-01 13:51:37] ERROR 19 [django.request] log.py:224 - [fd60f8a880f1] - Internal Server Error: /complete/saml/
web_1     | Traceback (most recent call last):
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
web_1     |     response = get_response(request)
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
web_1     |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/sentry_sdk/integrations/django/views.py", line 52, in callback
web_1     |     return old_callback(*args, **kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/sentry_sdk/integrations/django/views.py", line 52, in callback
web_1     |     return old_callback(*args, **kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
web_1     |     response = view_func(request, *args, **kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
web_1     |     return view_func(*args, **kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_django/utils.py", line 49, in wrapper
web_1     |     return func(request, backend, *args, **kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_django/views.py", line 31, in complete
web_1     |     return do_complete(request.backend, _do_login, user=request.user,
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_core/actions.py", line 87, in do_complete
web_1     |     url = setting_url(backend, 'INACTIVE_USER_URL', 'LOGIN_ERROR_URL',
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_core/utils.py", line 241, in setting_url
web_1     |     value = backend.setting(name)
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_core/backends/base.py", line 31, in setting
web_1     |     return self.strategy.setting(name, default=default, backend=self)
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_core/strategy.py", line 47, in setting
web_1     |     return self.get_setting(name)
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_django/strategy.py", line 48, in get_setting
web_1     |     value = resolve_url(value)
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/shortcuts.py", line 131, in resolve_url
web_1     |     return reverse(to, args=args, kwargs=kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/urls/base.py", line 87, in reverse
web_1     |     return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/urls/resolvers.py", line 685, in _reverse_with_prefix
web_1     |     raise NoReverseMatch(msg)
web_1     | django.urls.exceptions.NoReverseMatch: Reverse for 'login' not found. 'login' is not a valid view function or pattern name.
web_1     | [2021-04-01 13:51:37] ERROR 19 [django.request] log.py:224 - [fd60f8a880f1] - Internal Server Error: /complete/saml/
web_1     | Traceback (most recent call last):
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
web_1     |     response = get_response(request)
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
web_1     |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/sentry_sdk/integrations/django/views.py", line 52, in callback
web_1     |     return old_callback(*args, **kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/sentry_sdk/integrations/django/views.py", line 52, in callback
web_1     |     return old_callback(*args, **kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
web_1     |     response = view_func(request, *args, **kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
web_1     |     return view_func(*args, **kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_django/utils.py", line 49, in wrapper
web_1     |     return func(request, backend, *args, **kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_django/views.py", line 31, in complete
web_1     |     return do_complete(request.backend, _do_login, user=request.user,
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_core/actions.py", line 87, in do_complete
web_1     |     url = setting_url(backend, 'INACTIVE_USER_URL', 'LOGIN_ERROR_URL',
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_core/utils.py", line 241, in setting_url
web_1     |     value = backend.setting(name)
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_core/backends/base.py", line 31, in setting
web_1     |     return self.strategy.setting(name, default=default, backend=self)
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_core/strategy.py", line 47, in setting
web_1     |     return self.get_setting(name)
web_1     |   File "/usr/local/lib/python3.8/site-packages/social_django/strategy.py", line 48, in get_setting
web_1     |     value = resolve_url(value)
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/shortcuts.py", line 131, in resolve_url
web_1     |     return reverse(to, args=args, kwargs=kwargs)
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/urls/base.py", line 87, in reverse
web_1     |     return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
web_1     |   File "/usr/local/lib/python3.8/site-packages/django/urls/resolvers.py", line 685, in _reverse_with_prefix
web_1     |     raise NoReverseMatch(msg)
web_1     | django.urls.exceptions.NoReverseMatch: Reverse for 'login' not found. 'login' is not a valid view function or pattern name.
```

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -100,6 +100,7 @@ jobs:
           MAILGUN_SENDER_DOMAIN: 'fake.example.com'
           MAILGUN_KEY: 'fake'
           NODE_ENV: 'production'
+          SOCIAL_AUTH_SAML_LOGIN_URL: 'fake.example.com/login/saml/'
```
[the remark this answers is not in the document]
Comment (rhysyngsun) [reply .github/workflows/ci.yml:103]: This is the url on the IDP that the user goes to to login, for tests an arbitrary value is fine.

**Comment by rhysyngsun** [comment]
@mbertrand addressed your feedback, you shouldn't encounter that `NoReverseMatch` error now, but i think you were hitting that because social auth was trying to display _another_ error so we'll need to look into that tomorrow.

**Review by mbertrand — APPROVED** [review]
👍

**Comment by rhysyngsun** [comment]
@mbertrand I've updated the collaborator API and UI to use `User.id` instead of `User.username`

**Review by mbertrand — APPROVED** [review]
LGTM, just a pylint issue to fix 👍

**Pull request merged**